scripts/dev/e2e.py

```diff
@@ -7,6 +7,7 @@
     load_yaml_from_file,
 )
 import k8s_conditions
+import dump_diagnostic
```

Comment:
[nit] I think there are 2 different responsibilities being pulled into one. 

* Accessing/Fetching diagnostics about K8s resources
* Writing them to disk

I think it would be a nice to have a mechanism to access the resources separated from the act of writing them. What do you think?